### ISSUE TITLE
Boost not building due to b2 config error (Update: Added repro)

### ISSUE THREAD

**Issue by ColinGilbert** [ruslo/hunter#423]
Hi,

I am Hunterizing an existing project. Here is the error:
```
/home/noob/devel/hunter-root/_Base/xxxxxxx/de1723d/c05c545/Build/Boost/Source/tools/build/src/build/feature.jam:494: in feature.validate-value-string from module feature
error: "none" is not a known value of feature <optimization>
error: legal values: "off" "speed" "space"
```

Here are the relevant lines in CMakeLists.txt
```
cmake_minimum_required(VERSION 3.5)

include("HunterGate.cmake")

HunterGate(
            URL "https://github.com/ruslo/hunter/archive/v0.14.23.tar.gz"
                SHA1 "999f07d004649f5f549953f74f21e46b48d53931"
                )

project(NoobWerkz)

hunter_add_package(Boost COMPONENTS system filesystem)
find_package(Boost CONFIG REQUIRED system filesystem)

( ... more non-boost related lines ... )

target_link_libraries(NoobEdit LINK_PUBLIC ${OPENGL_LIBRARIES} freetype rt dl pthread m Boost::boost Boost::system Boost::filesystem )
```

My HunterGate.cmake file is copy+pasted from the current version available. Any idea what is wrong here?

EDIT: Here is the set of invocations used to trigger the build
```
cmake -H. -B_builds -DHUNTER_STATUS_DEBUG=ON  -DCMAKE_TOOLCHAIN_FILE=$POLLY_ROOT/gcc.cmake &&
cmake --build _builds --config Release
```

**Comment by ColinGilbert** [comment]
I made a reproduction of this problem with full error log in README: https://github.com/ColinGilbert/hunter-repro

**Comment by pretyman** [comment]
What platform/compiler are you using? I've cloned your project, ran the given command line and boost builded successfully. I'm on OSX using the XCode 7.3 Clang

**Comment by ColinGilbert** [comment]
I fixed the missing dash and retried. No difference; must have been a simple typo.

I am running CMake 3.5.2 on Gentoo Linux, gcc 4.9.3. The error appears due to the optimization level not being set during the boost build configure phase. If you want a full snapshot of all versions of software on my current development system please ask.

The Hunter-simple example works fine, btw.

**Comment by ruslo** [comment]
@ColinGilbert According to the logs it is a Boost configuration error. Try to download official release version 1.61.0 and run `./bootstap && ./b2 --layout=tagged -d0 install --without-atomic --without-chrono --without-container --without-context --without-coroutine --without-coroutine2 --without-date_time --without-exception --without-filesystem --without-graph --without-graph_parallel --without-iostreams --without-locale --without-log --without-math --without-metaparse --without-mpi --without-program_options --without-python --without-random --without-regex --without-serialization --without-signals --without-system --without-test --without-thread --without-timer --without-type_erasure --without-wave`

In case you will have same error you should open bug report to Boost. Feel free to add a link here.

**Comment by ruslo** [comment]
Boost bug, closing.

**Comment by Vlooe** [comment]
I can reproduce the problem and i think this is not a boost bug.
Gentoo provides /usr/share/boost-build/site-config.jam with this content:

```
# Copyright 1999-2012 Gentoo Foundation
# Distributed under the Boost Software License, Version 1.0.
# (See accompanying file LICENSE_1_0.txt or http://www.boost.org/LICENSE_1_0.txt)
 
# Define two new variants to be used when building boost (or separate boost-libs)
# on Gentoo. The two variants make use of Gentoo-specific optimization and debug-symbols
# values "none" which are not part of the official boost distribution.
# DO NOT RELY ON THE FOLLOWING VARIANTS TO BE PRESENT ON OTHER OS!
variant gentoorelease : release : <optimization>none <debug-symbols>none <runtime-link>shared ;
variant gentoodebug   : debug   : <optimization>none <debug-symbols>on   <runtime-link>shared ;
```

When i comment these lines out i do not get the 
`error: "none" is not a known value of feature <optimization>`

http://stackoverflow.com/questions/23013433/how-to-install-modular-boost
According to this post it could be solved by building with
`./b2 --ignore-site-config `

**Comment by ruslo** [comment]
> According to this post it could be solved by building with

Feel free to send pull request, need to test it.

**Comment by red-0ne** [comment]
Any news on when @Vlooe's fix will be merged?

**Comment by ruslo** [comment]
@red-0ne To merge pull request somebody have to open it first, no? :)

**Comment by red-0ne** [comment]
Oops thought is was already done.

**Comment by Vlooe** [comment]
Sry forgot to do the pull request :-/
But now its done :D

**Comment by red-0ne** [comment]
Excellent. Thank you.

**Comment by llitz** [comment]
Is that the only place where we need to place the --ignore-site-config ? I came across a different package, updated to the 0.19.27 Hunter version and I still have the same "none" error. Unfortunately, I don't understand enough about Hunter to figure it out on my own, any directions would be appreciated

**Comment by ruslo** [comment]
> I still have the same "none" error

I haven't experienced this error in the first place so I can't say anything...

@Vlooe any comments?

**Comment by llitz** [comment]
If anyone can point me towards the right place to fix this, I can add the ignore-site-config on the right place and report back.

This is my error:

/var/tmp/portage/net-misc/cpp-ethereum-9999/homedir/.hunter/_Base/0dd355c/96901c7/a7a04ee/Build/Boost/__system/Source/tools/build/src/build/feature.jam:494: in feature.validate-va
lue-string from module feature
error: "none" is not a known value of feature <optimization>
error: legal values: "off" "speed" "space"
/var/tmp/portage/net-misc/cpp-ethereum-9999/homedir/.hunter/_Base/0dd355c/96901c7/a7a04ee/Build/Boost/__system/Source/tools/build/src/build/property.jam:276: in validate1 from mod
ule property
/var/tmp/portage/net-misc/cpp-ethereum-9999/homedir/.hunter/_Base/0dd355c/96901c7/a7a04ee/Build/Boost/__system/Source/tools/build/src/build/property.jam:302: in property.validate
from module property
/var/tmp/portage/net-misc/cpp-ethereum-9999/homedir/.hunter/_Base/0dd355c/96901c7/a7a04ee/Build/Boost/__system/Source/tools/build/src/tools/builtin.jam:381: in variant from module
 builtin
/usr/share/boost-build/site-config.jam:9: in modules.load from module site-config
/var/tmp/portage/net-misc/cpp-ethereum-9999/homedir/.hunter/_Base/0dd355c/96901c7/a7a04ee/Build/Boost/__system/Source/tools/build/src/build-system.jam:249: in load-config from mod
ule build-system
/var/tmp/portage/net-misc/cpp-ethereum-9999/homedir/.hunter/_Base/0dd355c/96901c7/a7a04ee/Build/Boost/__system/Source/tools/build/src/build-system.jam:351: in load-configuration-f
iles from module build-system
/var/tmp/portage/net-misc/cpp-ethereum-9999/homedir/.hunter/_Base/0dd355c/96901c7/a7a04ee/Build/Boost/__system/Source/tools/build/src/build-system.jam:524: in load from module bui
ld-system
/var/tmp/portage/net-misc/cpp-ethereum-9999/homedir/.hunter/_Base/0dd355c/96901c7/a7a04ee/Build/Boost/__system/Source/tools/build/src/kernel/modules.jam:295: in import from module
 modules
/var/tmp/portage/net-misc/cpp-ethereum-9999/homedir/.hunter/_Base/0dd355c/96901c7/a7a04ee/Build/Boost/__system/Source/tools/build/src/kernel/bootstrap.jam:139: in boost-build from
 module
/var/tmp/portage/net-misc/cpp-ethereum-9999/homedir/.hunter/_Base/0dd355c/96901c7/a7a04ee/Build/Boost/__system/Source/boost-build.jam:17: in module scope from module

**Comment by ruslo** [comment]
> This is my error

Looks like boost.system is building. It means that `url_sha1_boost_library` was used:
* https://github.com/ruslo/hunter/blob/master/cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in

According to @Vlooe comment:
```
url_sha1_boost_library.cmake.in is generating its own .jam file. Thus the site config is not used
```
* https://github.com/ruslo/hunter/pull/863#discussion_r125251276

**Comment by llitz** [comment]
I've got it working. Had to add the --ignore-site-config in two other places insice the url_sha1_boost_library as well (I've tried reverting the modification and it didn't work).

Fixed with 09698d2

**Comment by Vlooe** [comment]
Hmm this happens if you do not double check/test the stuff you do :-/
Seems the site-config is used even if a user-config is used. Just confirmed it myself.
Do not remember what i did in May when i came to the conclusion that its not needed in url_sha1_boost_library.cmake.in
